diff --git a/src/lib.typ b/src/lib.typ
index 4679550..a72158d 100644
--- a/src/lib.typ
+++ b/src/lib.typ
@@ -150,6 +150,14 @@
   end-citation-group()
 }
 
+#let bibs = state("__bibs", ())
+#let add-bib(file) = context bibs.update(old => {
+  if old == none { old = (file) }
+  else { old.push(file) }
+  old
+})
+#let get-bibs() = bibs.final()
+
 /// Loads the bibliography for a given prefix.
 /// The function reads the bibliography from the given file(s), which is used later by
 /// @@get-bibliography(). It does not render any content.
@@ -168,10 +176,6 @@
 ///
 /// -> content
 #let load-bibliography(
-  /// the path(s) to the bibliography file(s), which is passed to the `read` function
-  /// registered via @@alexandria(), or its binary contents if no `read` function provided.
-  /// -> string | bytes | array
-  path,
   /// the optional prefix for which the bibliography is loaded.
   /// It only needs to be specified if more than one prefix was registered.
   /// -> string | auto
@@ -189,11 +193,6 @@
   import "state.typ": *
   import "internal.typ": *
 
-  let path = path
-  if type(path) != array {
-    path = (path,)
-  }
-
   context {
     let prefix = prefix
     if prefix == auto {
@@ -201,7 +200,7 @@
       assert.ne(prefix, none, message: "when using multiple custom bibliographies, you must specify the prefix for each")
     }
 
-    let sources = path.map(path => read(path))
+    let sources = get-bibs().map(path => read(path))
 
     let style = csl-to-string(style)
     if style in hayagriva.names {
